File: DSA_python/pascals-triangle.py
numRow = 5

ans = []


for i in range(numRow):
	sub = [1]

	if i > 0:

		prev = ans[i - 1]

		for j in range(1, i):
			sub.append(prev[j - 1] + prev[j])

		sub.append(1)

	ans.append(sub)


print(ans)

File: DSA_python/pascals-triangle.java
import java.util.*;

class triangle {
    public static void main(String args[]) {

        List<List<Integer>> ans = new ArrayList<>();

        int numRow = 5;

        for (int i = 0; i < numRow; i++) {

            List<Integer> sub = new ArrayList<>();
            sub.add(1);

            if (i > 0) {
                List<Integer> prev = ans.get(i - 1);

                for (int j = 1; j < i; j++) {
                    sub.add(prev.get(j - 1) + prev.get(j));
                }

                sub.add(1);
            }

            ans.add(sub);
        }

        System.out.println(ans);
    }
}